src/org/traccar/smpp/SmppClient.java

```diff
@@ -0,0 +1,218 @@
+    private byte destNpi;
+
+    public SmppClient() {
+        sessionConfig.setName("Traccar.smppSession");
+        sessionConfig.setInterfaceVersion(
+                (byte) Context.getConfig().getInteger("sms.smpp.version", SmppConstants.VERSION_3_4));
+        sessionConfig.setType(SmppBindType.TRANSCEIVER);
+        sessionConfig.setHost(Context.getConfig().getString("sms.smpp.host", "localhost"));
+        sessionConfig.setPort(Context.getConfig().getInteger("sms.smpp.port", 2775));
+        sessionConfig.setSystemId(Context.getConfig().getString("sms.smpp.username", "user"));
+        sessionConfig.setPassword(Context.getConfig().getString("sms.smpp.password", "password"));
+        sessionConfig.getLoggingOptions().setLogBytes(false);
+        sessionConfig.getLoggingOptions().setLogPdu(Context.getConfig().getBoolean("sms.smpp.logPdu"));
+
+        sourceAddress = Context.getConfig().getString("sms.smpp.sourceAddress", "");
+        submitTimeout = Context.getConfig().getInteger("sms.smpp.submitTimeout", 10000);
+
+        charsetName = Context.getConfig().getString("sms.smpp.charsetName", CharsetUtil.NAME_UCS_2);
+        dataCoding = (byte) Context.getConfig().getInteger("sms.smpp.dataCoding", SmppConstants.DATA_CODING_UCS2);
+
+        sourceTon = (byte) Context.getConfig().getInteger("sms.smpp.sourceTon", SmppConstants.TON_ALPHANUMERIC);
+        sourceNpi = (byte) Context.getConfig().getInteger("sms.smpp.sourceNpi", SmppConstants.NPI_UNKNOWN);
+
+        destTon = (byte) Context.getConfig().getInteger("sms.smpp.destTon", SmppConstants.TON_INTERNATIONAL);
+        destNpi = (byte) Context.getConfig().getInteger("sms.smpp.destNpi", SmppConstants.NPI_E164);
+
+        enquireLinkPeriod = Context.getConfig().getInteger("sms.smpp.enquireLinkPeriod", 60000);
+        enquireLinkTimeout = Context.getConfig().getInteger("sms.smpp.enquireLinkTimeout", 10000);
+        enquireLinkExecutor = Executors.newScheduledThreadPool(1, new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable runnable) {
+                Thread thread = new Thread(runnable);
+                String name = sessionConfig.getName();
+                thread.setName("EnquireLink-" + name);
+                return thread;
+            }
+        });
+
+        reconnectionDelay = Context.getConfig().getInteger("sms.smpp.reconnectionDelay", 10000);
+        reconnectionExecutor = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable runnable) {
+                Thread thread = new Thread(runnable);
+                String name = sessionConfig.getName();
+                thread.setName("Reconnection-" + name);
+                return thread;
+            }
+        });
+
+        scheduleReconnect();
+    }
+
+    public synchronized SmppSession getSession() {
+        return smppSession;
+    }
+
+    public String mapDataCodingToCharset(byte dataCoding) {
```

Comment:
This function is not full, there is a problem to map `dataCoding` to charsets https://github.com/twitter/cloudhopper-commons/issues/7